olive/cli/generate_adapter.py

```diff
@@ -0,0 +1,141 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from typing import ClassVar, Dict
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_logging_options,
+    add_model_options,
+    add_remote_options,
+    get_input_model_config,
+    is_remote_run,
+    save_output_model,
+    update_remote_option,
+)
+from olive.common.utils import set_nested_dict_value, set_tempdir
+
+
+class GenerateAdapterCommand(BaseOliveCLICommand):
+    allow_unknown_args: ClassVar[bool] = True
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "generate-adapter", help="Convert and optimize the model for ONNX Runtime with adapters as inputs"
+        )
+
+        add_logging_options(sub_parser)
+
+        sub_parser.add_argument(
+            "--precision",
+            type=str,
+            default="float16",
+            choices=["float16", "float32"],
+            help="The precision of the optimized model and adapters.",
+        )
+
+        # Model options
+        add_model_options(sub_parser, enable_hf=True, enable_hf_adapter=True)
```

Comment:
This CLI should also accept ONNX model as an input. In that case, skip onnx graph capturing & optimizations and just run extract adapter pass. MB metadata should be generated based on use-ort-genai flag, as it is right now.

Reply:
I will add this part. Just saw in the design doc that this is supported scenario